code/src/main/java/frc/robot/commands/MoveToLocationCommand.java

```diff
@@ -0,0 +1,39 @@
+package frc.robot.commands;
+
+import com.google.inject.Inject;
+
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.geometry.Translation2d;
+import frc.robot.RobotConfig;
+import frc.robot.common.IDrivetrainSubsystem;
+import frc.robot.common.ILogger;
+import frc.robot.common.TraceableCommand;
+
+public class MoveToLocationCommand extends TraceableCommand {
+    private Pose2d[] path;
+    private IDrivetrainSubsystem m_subsystem;
+    private OdometryCommand m_command;
+
+    @Inject
+    public MoveToLocationCommand(ILogger logger, IDrivetrainSubsystem subsystem, RobotConfig config, Pose2d position) {
```

Comment:
the literals are not going to work well with an IoC. The pattern would be to have these added as properties. Then what ever wanted this class would set the properties at the point of scheduling.